bnb-chain/bsc
# BEP-439: Implement EIP-2537: Precompile for BLS12-381 curve operations

### Description

[BEP-439: Implement EIP-2537: Precompile for BLS12-381 curve operations](https://github.com/bnb-chain/BEPs/pull/439)

### Rationale

tell us why we need these changes...

### Example

how to verify the code implement? two steps:
1. ensure test case files in `bsc/core/vm/testdata/precompiles` are the same with `go-ethereum/core/vm/testdata/precompiles` at the latest commit
2. run the test cases
```
cd core/vm
go test -run TestPrecompiledBLS12381
```
all pass


### Changes

the first two commitss are picked from upstream
the third commit refers to https://github.com/ethereum/go-ethereum/pull/29552
